lib/bolt/pal.rb

```diff
@@ -15,25 +15,33 @@ class PAL
     # PALError is used to convert errors from executing puppet code into
     # Bolt::Errors
     class PALError < Bolt::Error
-      # Puppet sometimes rescues exceptions notes the location and reraises.
-      # Return the original error.
       def self.from_preformatted_error(err)
         if err.cause&.is_a? Bolt::Error
           err.cause
         else
-          from_error(err.cause || err)
+          from_error(err)
         end
       end
 
       # Generate a Bolt::Pal::PALError for non-bolt errors
       def self.from_error(err)
-        e = new(err.message)
+        # Provide the location of an error if it came from a plan
+        details = if defined?(err.file)
```

Comment:
https://github.com/puppetlabs/bolt/blob/master/lib/bolt/pal.rb#L162 It looks like we have file/line/column in this message already. I like this format better, so I think removing that part of the message would be good for consistency + deduplication.

Reply:
More directly relevant to this line specifically, it looks like this will add these details if `err.file` is defined on the class, even if it and the line and position are nil:
```
lucy ~/githubs/bolt (1932/plan-errors) $ be bolt plan run bolt
Starting: plan bolt
Finished: plan bolt in 0.01 sec
{
  "kind": "bolt/pal-error",
  "msg": "Function lookup() did not find a value for the name 'abcdef'",
  "details": {
    "file": null,
    "line": null,
    "column": null
  }
}
```

It might be better to ensure this exists *and* isn't nil. It actually looks like these might be set separately, so we could theoretically just add each item to the hash if it isn't nil, but having a line and column without a file name seems pretty useless so I'm inclined to just not add the details if the file is nil.

Reply:
Updated. If these changes cover what we want I'll squash the commits. :)